sql/core/src/test/scala/org/apache/spark/sql/columnar/ColumnStatsSuite.scala

```diff
@@ -22,19 +22,19 @@ import org.apache.spark.sql.catalyst.expressions.InternalRow
 import org.apache.spark.sql.types._
 
 class ColumnStatsSuite extends SparkFunSuite {
+  testColumnStats(classOf[BooleanColumnStats], BOOLEAN, InternalRow(true, false, 0))
```

Comment:
By re-ordering catalyst date type handling, I found we missed boolean type here. Added.
However, we still miss binary type here, as `BINARY` is not subclass of `NativeColumnType` but `testColumnStats` only accepts `NativeColumnType`.
I'm wondering if we could make `BINARY` extend `NativeColumnType`, as `BinaryType` extends `AtomicType`.
cc @liancheng